plugin/storage/es/spanstore/reader.go

```diff
@@ -421,12 +425,17 @@ func (s *SpanReader) buildFindTraceIDsQuery(traceQuery *spanstore.TraceQueryPara
 	startTimeQuery := s.buildStartTimeQuery(traceQuery.StartTimeMin, traceQuery.StartTimeMax)
 	boolQuery.Must(startTimeQuery)
 
-	//add process.serviceName query
-	if traceQuery.ServiceName != "" {
+	//add process.serviceName query, only if AnyServiceName is blank
+	if traceQuery.ServiceName != "" && !traceQuery.AnyServiceName {
```

Comment:
I would prefer to return an error when both `traceQuery.ServiceName` and `traceQuery.AnyServiceName` are non-empty